### ISSUE TITLE
Undefined index has_children in widget-conditions.php

### ISSUE THREAD

**Issue by chuckreynolds** [Automattic/jetpack#3070]
Trying to track this error down when loading /wp-admin/widgets.php.

```text
PHP Notice:  Undefined index: has_children in /wp-content/plugins/jetpack/modules/widget-visibility/widget-conditions.php on line 282
PHP Stack trace:
PHP   1. {main}() /wp-admin/widgets.php:0
PHP   2. wp_list_widget_controls() /wp-admin/widgets.php:383
PHP   3. dynamic_sidebar() /wp-admin/includes/widgets.php:97
PHP   4. call_user_func_array:{/wp-includes/widgets.php:1319}() /wp-includes/widgets.php:1319
PHP   5. wp_widget_control() /wp-includes/widgets.php:1319
PHP   6. call_user_func_array:{/wp-admin/includes/widgets.php:232}() /wp-admin/includes/widgets.php:232
PHP   7. WP_Widget->form_callback() /wp-admin/includes/widgets.php:232
PHP   8. do_action_ref_array() /wp-includes/widgets.php:494
PHP   9. call_user_func_array:{/wp-includes/plugin.php:579}() /wp-includes/plugin.php:579
PHP  10. Jetpack_Widget_Conditions::widget_conditions_admin() /wp-includes/plugin.php:579
```

WordPress v4.3.1
Jetpack v3.8.0

I know that's not a great bug report but not familiar w/ the plugin/functionality.

**Comment by kraftbj** [comment]
Still helpful!

L282 moved to L290 https://github.com/Automattic/jetpack/blob/3.8.1-beta3/modules/widget-visibility/widget-conditions.php#L290

Sounds like `$instance['conditions']` has `rules` already set, so no empty options are being added https://github.com/Automattic/jetpack/blob/3.8.1-beta3/modules/widget-visibility/widget-conditions.php#L246 , but the conditions being passed lack the expected key.

@chuckreynolds Have you been using Widget Visibility for awhile? My initial hypothesis is something along the lines that a previously saved rule was added before `has_children` existed, and thus never updated.

**Comment by jeherve** [comment]
Fixed in #3286

**Comment by chuckreynolds** [comment]
:+1: sounds like #3286 should fix it. Yes @kraftbj that site has used widget visibility for quite some time, since before my time. The DB is all sorts of dirty so having that exception and/or cleanup function will help. Thx guys. Cheers.
